### PR TITLE
Voting results

### PR DESCRIPTION
**Pull request by trumko**

### CODE REVIEW

File: server/sockets.js [hunk 1]
```diff
@@ -16,11 +16,14 @@ module.exports = (server) => {
     });
 
     socket.on('startTimer', (roomId) => {
-      var timeCount = 21;
+      var timeCount = 2;
```
Comment (ktv18) [inline server/sockets.js:19]: let instead of var

File: server/sockets.js [hunk 2]
```diff
@@ -16,11 +16,14 @@ module.exports = (server) => {
...
+
+      io.sockets.emit('dropValues');
 
       Room.findById(roomId, (err, room) => {
         room.results = [];
         room.save();
+        console.log(room)
```
Comment (ktv18) [inline server/sockets.js]: do you really need it ?

File: server/sockets.js [hunk 3]
```diff
@@ -43,7 +46,12 @@ module.exports = (server) => {
         room.results.push(userChoise);
 
         room.save((err, changedRoom) => {
-          //TODO: Compare room.results with room.users and emit event when they are equal
+          Room.findById(userInfo.roomId, (err, room) => {
+            if (room.results.length == room.users.length) {
```
Comment (ktv18) [inline server/sockets.js]: use `===`

File: server/sockets.js [hunk 4]
```diff
@@ -43,7 +46,12 @@ module.exports = (server) => {
...
+              console.log('they are equal');
```
Comment (ktv18) [inline server/sockets.js]: again console

File: src/components/elements/lists/UserItem.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
+class UserItem extends React.Component {
+
+  render() {
+    let isRotated = this.props.value ? 'flip-container-rotated' : null;
+    console.log('asd', this.props.extraValues);
```
Comment (ktv18) [inline src/components/elements/lists/UserItem.js]: again

File: src/components/elements/lists/UserItem.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    let maxValue = this.props.value === this.props.extraValues.max ? 'max-value' : null
+    let minValue = this.props.value === this.props.extraValues.min ? 'min-value' : null
+    console.log(maxValue, minValue)
```
Comment (ktv18) [inline src/components/elements/lists/UserItem.js]: again

File: src/components/elements/lists/UserItem.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        		<div className="back">
+              {this.props.value}
+              {maxValue} {minValue}
+        		</div>
+        	</div>
+        </div>
+        <div>{this.props.userName}</div>
+      </li>
+    )
+  }
```
Comment (ktv18) [inline src/components/elements/lists/UserItem.js:23]: strange formatting inside return block

File: src/components/layouts/UsersList.js [hunk 8]
```diff
@@ -2,36 +2,50 @@ import React from 'react';
...
     this.props.socket.on('send_room', data => {
       this.props.addUsers(data.users);
     });
 
     this.props.socket.on('userDisconnect', data => {
       this.props.removeUser(data.userId);
     });
+
+    this.props.socket.on('showResults', data => {
+      let results = data.map(item => item.value).filter(item => item !== -1)
```
Comment (ktv18) [inline src/components/layouts/UsersList.js]: could you please rewrite it by using one loop ? (reduce)

File: src/reducers/userReducer.js [hunk 9]
```diff
@@ -6,17 +6,35 @@ let initialState = {
...
+    case ActionTypes.DROP_VALUES:
+      newUsers = JSON.parse(JSON.stringify(state.users));
+      newUsers.forEach(elem => {
+        elem.value = null;
+      })
+      return Object.assign({}, state, { users: newUsers });
     case ActionTypes.REMOVE_USER:
       let userIdx = state.users.map(elem => elem.id).indexOf(action.userId);
-      let newUsers = JSON.parse(JSON.stringify(state.users));
+      newUsers = JSON.parse(JSON.stringify(state.users));
```
Comment (ktv18) [inline src/reducers/userReducer.js:34]: intresting, do you just copy objects ?

Comment (trumko) [reply src/reducers/userReducer.js:34]: Yes, we used it for deep coping of object, maybe you know a better way how to do this?

Comment (ktv18) [reply src/reducers/userReducer.js:34]: i think this way should be fine unless state.users has methods
also you can use Object assign i guess, anyway it's fine

File: server/sockets.js (hunk 1 above)
Comment (trumko) [reply server/sockets.js:19]: We used ```var``` because of our version of Node.js

Comment (ktv18) [reply server/sockets.js:19]: which version do you use and did you try `let` ?

**Review by ktv18 — APPROVED** [review]
r + a

**Pull request merged**